### PR TITLE
Generate LICENSES, make --branch required

### PR DESCRIPTION
**Pull request by ostromart**
Couple of items in this PR:
 - make --branch mandatory so that LICENSES URLs point to license text on correct branch.
 - initial output of LICENSES for 1.0 release

License notes:
1. github.com/signalfx/com_signalfx_metrics_protobuf is only used in testing. In general we need some automated mechanism to ignore anything that doesn't end up in a binary (https://github.com/istio/istio/issues/7288). 
2. siphash is used in open-policy-agent/opa. The version we have in 1.0 uses 0.7.1 which still points to the old siphash which doesn't have a license. The new one is 0.8.2 which does have a license (https://github.com/istio/istio/issues/7287)
3. github.com/BurntSushi/toml has WTFPL at the version in vendor. This is part of istioctl through k8s.io/helm which is at 2.8.2. Need 2.9.1 which gets rid of this pkg (https://github.com/istio/istio/issues/7293).

### CODE REVIEW

**Comment by rshriram** [comment]
Do we need to include this file? its 5.3MB!

**Comment by rshriram** [comment]
I think we should generate it during the release and not commit it in our repo. Its huge and unnecessary. Out of curiosity, are there any GPL style licenses in there?

File: tools/license/get_dep_licenses.go [hunk 1]
```diff
@@ -16,14 +16,14 @@
 // This tool requires https://github.com/benbalter/licensee to work.
 // Usage:
 //   1) Generate complete dump of every license, suitable for including in release build/binary image:
-//      go run get_dep_licenses.go
+//      go run get_dep_licenses.go --branch release-0.8
```
Comment (andraxylia) [inline tools/license/get_dep_licenses.go]: release-1.0?

**Comment by ostromart** [comment]
@rshriram I think we are ok with the file size. It has huge redundancy so it zips to ~97k. git will compress it automatically per https://gist.github.com/matthewmccullough/2695758.

**Pull request merged**